### ISSUE TITLE
[QUESTION] - How do I get it running?

### ISSUE THREAD

**Issue by is-it-ayush** [veeso/remotefs-rs#27]
Hi @veeso, I am having trouble getting this library to work. My development environment is gnu/linux (debian 12) and I develop primarily for windows. I followed the installation steps in the readme. Although, I couldn't get it working. Here is the dependencies I added to my `cargo.toml`.
```
[dependencies]
remotefs = "^0.2.0"
remotefs-smb = "^0.1.0"
```
Here is the log dump of rust-analyzer,
<details>
<summary>rust-analyzer log dump</summary>

```
[ERROR rust_analyzer::main_loop] FetchBuildDataError:
error: failed to run custom build command for `pavao v0.1.2`

Caused by:
  process didn't exit successfully: `/home/ayush/personal/rust-ns/target/debug/build/pavao-a627555540e6bd34/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-env-changed=SMBCLIENT_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS_x86_64-pc-windows-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS_x86_64_pc_windows_gnu
  cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_ALLOW_CROSS
  cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-pc-windows-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_pc_windows_gnu
  cargo:rerun-if-env-changed=TARGET_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-pc-windows-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_pc_windows_gnu
  cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  error: SMB Client library not found! Probably libsmbclient is not installed.

  --- stderr
  thread 'main' panicked at 'pkg-config has not been configured to support cross-compilation.

  Install a sysroot for the target platform and configure it via
  PKG_CONFIG_SYSROOT_DIR and PKG_CONFIG_PATH, or install a
  cross-compiling wrapper for pkg-config and set it via
  PKG_CONFIG environment variable.', /home/ayush/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pavao-0.1.2/build.rs:14:13
  stack backtrace:
     0: rust_begin_unwind
               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/panicking.rs:593:5
     1: core::panicking::panic_fmt
               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/core/src/panicking.rs:67:14
     2: core::panicking::panic_display
               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/core/src/panicking.rs:150:5
     3: build_script_build::main
               at ./build.rs:14:13
     4: core::ops::function::FnOnce::call_once
               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/core/src/ops/function.rs:250:5
  note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```
</details>

I already ran `apt install -y libsmbclient-dev libsmbclient` and have these dependencies installed on my debian machine (from what I understand I shouldn't even need them for windows since [iirc you mentioned](https://github.com/veeso/remotefs-rs-smb/issues/1#issuecomment-1595772674) it uses windows native api's). My `.cargo/config.toml` contains the following,
```
[build]
target="x86_64-pc-windows-gnu"
```
Let me know if I missed a step. It'd be a great help since I can't get the rust-analyzer to load & the project doesn't build either even with explicitly specifying `--target` flag! 🙂

**Comment by veeso** [comment]
`x86_64-pc-windows-gnu` This target is considered UNIX derived, so it will try to build with libsmbclient

**Comment by is-it-ayush** [comment]
Interesting! I switched the target to `x86_64-pc-windows-msvc`. It still fails. Here is the log dump,

<details>

```sh
➜  rust-ns git:(master) ✗ sudo cargo build -vvv
       Fresh unicode-ident v1.0.11
       Fresh proc-macro2 v1.0.66
       Fresh autocfg v1.1.0
       Fresh quote v1.0.33
       Fresh libc v0.2.147
       Fresh syn v2.0.29
       Fresh winapi v0.3.9
       Fresh thiserror-impl v1.0.47
       Fresh pkg-config v0.3.27
       Fresh thiserror v1.0.47
       Fresh num-traits v0.2.16
       Fresh time v0.1.45
       Fresh log v0.4.20
       Fresh chrono v0.4.26
   Compiling pavao v0.1.2
       Fresh lazy_static v1.4.0
       Fresh wildmatch v2.1.1
       Fresh remotefs v0.2.0
     Running `/home/ayush/personal/rust-ns/target/debug/build/pavao-a627555540e6bd34/build-script-build`
[pavao 0.1.2] cargo:rerun-if-env-changed=SMBCLIENT_NO_PKG_CONFIG
[pavao 0.1.2] cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS_x86_64-pc-windows-msvc
[pavao 0.1.2] cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS_x86_64_pc_windows_msvc
[pavao 0.1.2] cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_ALLOW_CROSS
[pavao 0.1.2] cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS
[pavao 0.1.2] cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-pc-windows-msvc
[pavao 0.1.2] cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_pc_windows_msvc
[pavao 0.1.2] cargo:rerun-if-env-changed=TARGET_PKG_CONFIG
[pavao 0.1.2] cargo:rerun-if-env-changed=PKG_CONFIG
[pavao 0.1.2] cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-pc-windows-msvc
[pavao 0.1.2] cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_pc_windows_msvc
[pavao 0.1.2] cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_SYSROOT_DIR
[pavao 0.1.2] cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
[pavao 0.1.2] thread 'main' panicked at 'pkg-config has not been configured to support cross-compilation.
[pavao 0.1.2]
[pavao 0.1.2] Install a sysroot for the target platform and configure it via
[pavao 0.1.2] PKG_CONFIG_SYSROOT_DIR and PKG_CONFIG_PATH, or install a
[pavao 0.1.2] cross-compiling wrapper for pkg-config and set it via
[pavao 0.1.2] PKG_CONFIG environment variable.', /home/ayush/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pavao-0.1.2/build.rs:14:13
[pavao 0.1.2] note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[pavao 0.1.2] error: SMB Client library not found! Probably libsmbclient is not installed.
error: failed to run custom build command for `pavao v0.1.2`

Caused by:
  process didn't exit successfully: `/home/ayush/personal/rust-ns/target/debug/build/pavao-a627555540e6bd34/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-env-changed=SMBCLIENT_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS_x86_64-pc-windows-msvc
  cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS_x86_64_pc_windows_msvc
  cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_ALLOW_CROSS
  cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-pc-windows-msvc
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_pc_windows_msvc
  cargo:rerun-if-env-changed=TARGET_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-pc-windows-msvc
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_pc_windows_msvc
  cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  error: SMB Client library not found! Probably libsmbclient is not installed.

  --- stderr
  thread 'main' panicked at 'pkg-config has not been configured to support cross-compilation.

  Install a sysroot for the target platform and configure it via
  PKG_CONFIG_SYSROOT_DIR and PKG_CONFIG_PATH, or install a
  cross-compiling wrapper for pkg-config and set it via
  PKG_CONFIG environment variable.', /home/ayush/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pavao-0.1.2/build.rs:14:13
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

</details>

**Comment by veeso** [comment]
Are you running `cargo build --target x86_64-pc-windows-msvc`?

**Comment by is-it-ayush** [comment]
Yep, Here's the output veeso. I ran this on my debian host machine.

<details>

```sh
➜  rust-ns git:(master) ✗ sudo cargo build --target x86_64-pc-windows-msvc
   Compiling pavao v0.1.2
error: failed to run custom build command for `pavao v0.1.2`

Caused by:
  process didn't exit successfully: `/home/ayush/personal/rust-ns/target/debug/build/pavao-a627555540e6bd34/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-env-changed=SMBCLIENT_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS_x86_64-pc-windows-msvc
  cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS_x86_64_pc_windows_msvc
  cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_ALLOW_CROSS
  cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-pc-windows-msvc
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_pc_windows_msvc
  cargo:rerun-if-env-changed=TARGET_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-pc-windows-msvc
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_pc_windows_msvc
  cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  error: SMB Client library not found! Probably libsmbclient is not installed.

  --- stderr
  thread 'main' panicked at 'pkg-config has not been configured to support cross-compilation.

  Install a sysroot for the target platform and configure it via
  PKG_CONFIG_SYSROOT_DIR and PKG_CONFIG_PATH, or install a
  cross-compiling wrapper for pkg-config and set it via
  PKG_CONFIG environment variable.', /home/ayush/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pavao-0.1.2/build.rs:14:13
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

</details>

**Comment by is-it-ayush** [comment]
Hi @veeso, I found this problem. Apparently I was using the "^0.1.0" for `remotefs-smb` client library. The caret (^) didn't help as Cargo determined this compatibility range as anything between >=0.1.0 but <0.2.0. Therefore I didn't had the latest version of `remotefs-rs-smb` client library.

The old version is still specified in the `README.md`. I wrote a PR updating the readme so future user's won't have a similar issue.
